src/mnode/mnode-messageproc.h

```diff
@@ -13,21 +13,30 @@ extern CCriticalSection cs_mapOurMessages;
 bool Sign(const std::string& message, std::string& signatureBase64, std::string& error_ret);
 bool Sign(const std::string& message, std::vector<unsigned char>& signature, std::string& error_ret);
 
+// Type to distinguish the way we build/parse messages.
+enum class CMasternodeMessageType
+{
+    PLAINTEXT,
+    SETFEE
+};
+
 class CMasternodeMessage
 {
 public:
     CTxIn vinMasternodeFrom;
     CTxIn vinMasternodeTo;
+    int messageType;
```

Comment:
why unsigned short? the default underlying type of an enum class is always int.
and why const with default value PLAINTEXT?

Reply:
unsigned short for less space in database, even better uint8_t
const will be set in the constructor and then will be const all time. 
You can skip PLAINTEXT as you set value in constructor but I set all time default values in the header.
It can be UNDEFINED value by default too.

You can even try:
enum class CMasternodeMessageType : uint8_t
{
    UNDEFINED = 0,
    PLAINTEXT,
    SETFEE
};
and declare CMasternodeMessageType messageType{CMasternodeMessageType::UNDEFINED};
and check if serialization will work without need casting

You have a default constructor so better to set some default UNDEFINED value in the header.

Reply:
will change to uint8_t, but won't change to const.